### PR TITLE
Add support for smart contract constructor parameters.

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why is this change needed?

Until now, we deployed all smart contracts without constructor params.
Also, there was no clear structure from a solidity contract to metadata accessible through golang

### What changes were made as part of this PR:

Functional PR (with some refactoring traces).

- added script files that compile solidity contracts 
- standardised the folder structure of the script output
- create a "contract.go" file for each contract that contains functions to retrieve the bytecode with constructor params
- modify the erc20 to take in constructor params
- modify the guessing game to take in constructor params

### What are the key areas to look at
```have a look at the sol file in each
and maybe the .sh file
and the contract.go
basically in contracts.go , in each of those folders there's a func that gives you the deploy bytecode including constructor params
the rest of the changes in the codebase basdically replace the old way of getting the bytecode with that func
and also , have a look at the tools, which is the stuff you're more familiar with, as I've included a way to add parameters for deploying
```

### CODE REVIEW

File: contracts/managementcontract/build_mgmt_contract.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+#
+# This script builds the guessing game from solidity to a go generated package
```
Comment (joeldudley) [inline contracts/managementcontract/build_mgmt_contract.sh:4]: @tudor-malene Is this comment a mistake? Should it refer to the management contract instead?

File: integration/erc20contract/build_erc20_contract.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+#
+# This script builds the guessing game from solidity to a go generated package
```
Comment (joeldudley) [inline integration/erc20contract/build_erc20_contract.sh:4]: @tudor-malene Same here - should there be a reference to the guessing game here?

File: tools/contractdeployer/contract_deployer.go [hunk 3]
```diff
@@ -5,37 +5,48 @@ import (
...
 const (
 	mgmtContract         = "MGMT"
 	erc20Contract        = "ERC20"
 	guessingGameContract = "GUESS"
-	timeoutWait          = 80 * time.Second
-	retryInterval        = 2 * time.Second
 )
 
-type ContractDeployer struct {
+const (
```
Comment (joeldudley) [inline tools/contractdeployer/contract_deployer.go:30]: @tudor-malene Should be a single const block?

File: contracts/managementcontract/build_mgmt_contract.sh (hunk 1 above)
Comment (tudor-malene) [reply contracts/managementcontract/build_mgmt_contract.sh:4]: copy pasta

File: tools/contractdeployer/contract_deployer.go (hunk 3 above)
Comment (tudor-malene) [reply tools/contractdeployer/contract_deployer.go:30]: I split it in 2, to separate the types of supported contracts (an enum in a normal language), from the other stuff

Comment (joeldudley) [reply tools/contractdeployer/contract_deployer.go:30]: Ah yeah
